crates/burn-train/src/learner/classification.rs

```diff
@@ -27,6 +27,23 @@ impl<B: Backend> Adaptor<LossInput<B>> for ClassificationOutput<B> {
     }
 }
 
+impl<B: Backend> Adaptor<PrecisionInput<B>> for ClassificationOutput<B> {
+    fn adapt(&self) -> PrecisionInput<B> {
+        let [_, num_classes] = self.output.dims();
+        if num_classes > 1 {
+            PrecisionInput::new(
+                self.output.clone(),
+                self.targets.clone().one_hot(num_classes).bool(),
+            )
+        } else {
+            PrecisionInput::new(
+                self.output.clone(),
+                self.targets.clone().unsqueeze_dim(1).bool(),
+            )
+        }
```

Comment:
Shouldn't this also transform the targets with `.one_hot(...)` but force the num classes to 2 (assuming binary classification)?

Reply:
Hum, I think that would be covered by the case above since then `num_classes == 2`. This, I think, is slightly different from binary classification, is more multi class with 2 classes. Finally, I don't think it would work as output and targets should have the same shape.

Reply:
Spam vs not spam (i.e., positive vs negative) is still considered binary classification. I think what you are talking about is just that one uses sigmoid to model the positive output (so there is only one score) vs softmax where you have a score for each. But the targets are still represented as 0 or 1.

I'm not sure what `self.targets.clone().unsqueeze_dim(1).bool()` is supposed to represent 🤔 I could be missing something though

Reply:
Sure, at the end of the day they all are the same thing. The point I was trying to make is that in binary classification tasks I expect the output of the model to be (batch_size x 1) while in multi class I would expect (batch_size x 2), thus the targets for the first would be transformed from (batch_size) -> (batch_size, 1) and the second (batch_size) -> (batch_size, 2) so they match the shapes of the outputs. Still this is a preference, if you think it doesn't make sense we can just assert that the second dim of output  is not less than 2.

Reply:
Ahhh ok nvm I thought this would lead to some issues because the targets are expected to be one-hot encoded, but in reality this is not entirely true for the binary case for a single scalar output and target. The operations performed will still be valid.